utils/utils.go

```diff
@@ -1058,26 +1057,36 @@ func ForkVersionAtEpoch(epoch uint64) *types.ForkVersion {
 	}
 }
 
-func LogError(infoIdentifier string, err error) *logrus.Entry {
-	var fileIdentifier, lineIdentifier, functionIdentifier string
+func LogFatal(err error, errorMsg interface{}, callerSkip int, infoIdentifiers ...string) {
+	logErrorInfo(err, callerSkip, infoIdentifiers...).Fatal(errorMsg)
+}
+
+func LogError(err error, errorMsg interface{}, callerSkip int, infoIdentifiers ...string) {
+	logErrorInfo(err, callerSkip, infoIdentifiers...).Error(errorMsg)
+}
 
-	pc, fullFilePath, line, ok := runtime.Caller(1)
+func logErrorInfo(err error, callerSkip int, infoIdentifiers ...string) *logrus.Entry {
+	logFields := logrus.NewEntry(logrus.New())
+
+	pc, fullFilePath, line, ok := runtime.Caller(callerSkip + 2)
 	if ok {
-		fileIdentifier = filepath.Base(fullFilePath)
-		functionIdentifier = runtime.FuncForPC(pc).Name()
-		lineIdentifier = strconv.Itoa(line)
+		logFields = logFields.WithFields(logrus.Fields{
+			"file":     filepath.Base(fullFilePath),
+			"line":     line,
+			"function": runtime.FuncForPC(pc).Name(),
```

Comment:
You are super lucky that the alphabetical order here is "file", "function", "line". This is exactly the order I would like the info to appear.

You are not so lucky that the additional info passed to this function is then labeled `info_X` (see below). This separates `line` from the rest of the call stack info.

Maybe
```suggestion
			"cs_file":     filepath.Base(fullFilePath),
			"cs_line":     line,
			"cs_function": runtime.FuncForPC(pc).Name(),
```
or something similar?

That way you would get `level`, `msg`, `call stack info`, `additional info`.

Reply:
I see, so it's in alphabetic order but only for the fields themselves with level and msg always being first.
I did as you suggested and also moved line down in the code.